tests/test_scalars.py

```diff
@@ -1325,13 +1323,17 @@ def test_timestamp_cmp() -> None:
     lt = check(assert_type(ts < c_dt_datetime, bool), bool)
     assert gte != lt
 
-    check(assert_type(ts >= c_datetimeindex, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(ts < c_datetimeindex, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(ts >= c_unknown_index, np_ndarray_bool), np.ndarray, np.bool_)
-    check(assert_type(ts < c_unknown_index, np_ndarray_bool), np.ndarray, np.bool_)
+    check(assert_type(ts >= c_datetimeindex, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ts < c_datetimeindex, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ts >= c_unknown_index, np_1darray[np.bool]), np_1darray[np.bool])
+    check(assert_type(ts < c_unknown_index, np_1darray[np.bool]), np_1darray[np.bool])
 
     check(assert_type(ts >= c_np_ndarray_dt64, np_ndarray_bool), np.ndarray, np.bool_)
     check(assert_type(ts < c_np_ndarray_dt64, np_ndarray_bool), np.ndarray, np.bool_)
+    check(
+        assert_type(c_np_2darray_dt64 >= ts, np_2darray[np.bool]), np_2darray[np.bool]
+    )
+    check(assert_type(c_np_2darray_dt64 < ts, np_2darray[np.bool]), np_2darray[np.bool])
```

Comment:
So I won't keep repeating here.  If the `np.bool_` argument was in `check()` before, it needs to be there now.

And if you still have `np.ndarray` in the `check` function, I think it should be changed to `np.1darray[np.bool]`

Have to do that throughout this test func

Reply:
> And if you still have `np.ndarray` in the `check` function, I think it should be changed to `np.1darray[np.bool]`

Same answer as above. We shouldn't abruptly consider all results to be 1D arrays. If you compare the timestamp to an array with arbitrary shape, you get arbitrary shaped array of bool, not 1D array.

Reply:
Right, but if pandas returns a 1D array, then we need to test that as well.  Suggested elsewhere.